cowprotocol/cowswap
# Warnings for unfillable orders

# Summary

Fixes #1354

1. Show a warning if the token's allowance is insufficient  for the limit-order filling
2. Show a warning if the token's balance is insufficient for the limit-order filling

These warnings should be displayed only for Open orders

![image](https://user-images.githubusercontent.com/7122625/202701456-148a8733-25e3-4444-9cce-1667377ead34.png)

![image](https://user-images.githubusercontent.com/7122625/202701465-272f53df-4ed0-49ed-84d2-6413e7e0a94e.png)

  # To Test

1. Create order 0.5 WETH -> 4000 COW
- [ ] WETH balance should be > 0.5
- [ ] There is no warning about balance insufficiency 
2. Decrease WETH balance bellow 0.5 (Sell WETH using Swap)
- [ ] There must be a warning about balance insufficiency
3. Create order 0.5 WETH -> 4000 COW
- [ ] WETH allowance should be > 0.5
- [ ] There is no warning about allowance insufficiency 
2. Decrease WETH allowance bellow 0.5 (Call token's smart-contract)
- [ ] There must be a warning about allowance insufficiency
